fly/commands/internal/flaghelpers/pipeline_flag.go

```diff
@@ -11,23 +12,71 @@ import (
 	"github.com/concourse/concourse/go-concourse/concourse"
 )
 
-type PipelineFlag string
+type PipelineFlag struct {
+	Name         string
+	InstanceVars atc.InstanceVars
+}
 
 func (flag *PipelineFlag) Validate() ([]concourse.ConfigWarning, error) {
-	if strings.Contains(string(*flag), "/") {
-		return nil, errors.New("pipeline name cannot contain '/'")
-	}
-
 	var warnings []concourse.ConfigWarning
-	if warning := atc.ValidateIdentifier(string(*flag), "pipeline"); warning != nil {
-		warnings = append(warnings, concourse.ConfigWarning{
-			Type:    warning.Type,
-			Message: warning.Message,
-		})
+	if flag != nil {
+		if strings.Contains(flag.Name, "/") {
+			return nil, errors.New("pipeline name cannot contain '/'")
+		}
+
+		if warning := atc.ValidateIdentifier(flag.Name, "pipeline"); warning != nil {
+			warnings = append(warnings, concourse.ConfigWarning{
+				Type:    warning.Type,
+				Message: warning.Message,
+			})
+		}
 	}
 	return warnings, nil
 }
 
+func (flag *PipelineFlag) Ref() atc.PipelineRef {
+	return atc.PipelineRef{Name: flag.Name, InstanceVars: flag.InstanceVars}
+}
+
+func (flag *PipelineFlag) UnmarshalFlag(value string) error {
+	if !strings.Contains(value, "/") {
+		flag.Name = value
+		return nil
+	}
+
+	vs := strings.SplitN(value, "/", 2)
+	if len(vs) == 2 {
+		flag.Name = vs[0]
+		flatInstanceVars, err := unmarshalDotNotation(vs[1])
+		if err != nil {
+			return err
+		}
+		flag.InstanceVars, err = flatInstanceVars.Expand()
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func unmarshalDotNotation(value string) (atc.DotNotation, error) {
+	dn := atc.DotNotation{}
+	for _, v := range strings.Split(value, ",") {
```

Comment:
It would also be nice to not *require* dot notation for querying, but that's not so obvious to me how to do that, so I'm fine with this